### ISSUE TITLE
Date to unix timestamp conversion issue in 1.6

### ISSUE THREAD

**Issue by jesperoman** [stedolan/jq#1982]
**Describe the bug**
```fromdateiso8601``` outputs a unix timestamp 1 hour in the future (a value 3600 seconds larger than expected)

**To Reproduce**
```bash
date -d '2019-10-04T10:37:14Z' -u +%s && \
  jq-1.5 -nr '"2019-10-04T10:37:14Z"|fromdateiso8601' && \
  jq-1.6 -nr '"2019-10-04T10:37:14Z"|fromdateiso8601'
1570185434
1570185434
1570189034
```

**Expected behavior**
jq 1.6 outputting the same timestamp as gnu date and jq 1.5 like this:
```bash
date -d '2019-10-04T10:37:14Z' -u +%s && \
  jq-1.5 -nr '"2019-10-04T10:37:14Z"|fromdateiso8601' && \
  jq-1.6 -nr '"2019-10-04T10:37:14Z"|fromdateiso8601'
1570185434
1570185434
1570185434
```

**Environment (please complete the following information):**
 - OS and Version: [macOS 10.13.6 (fish shell 3.0.2, bash 5.0.2, zsh 5.7.1) and Linux Ubuntu 16.04.6 LTS (zsh 5.0.8)]
 - jq version [1.6] (compared with jq 1.5 in all environments which works as expected)

**Additional context**
Nothing else needed I believe

**Comment by jesperoman** [comment]
Did some more testing, and it seems like the problem started at:
unix timestamp ```1553997600``` (```2019-03-31T02:00:00Z```)

Example:
```
jq-1.6 -nr '1553997599 as $d|$d == ($d|todateiso8601|fromdateiso8601)'
true
```
After that time, it breaks:
```
jq-1.6 -nr '1553997600 as $d|$d == ($d|todateiso8601|fromdateiso8601)'
false
```
But still works in jq 1.5:
```
jq-1.5 -nr '1553997600 as $d|$d == ($d|todateiso8601|fromdateiso8601)'
true
```
Seems like todateiso8601 works as it should though:
```
jq-1.6 -nr '1553997600 as $d|$d|todateiso8601'
2019-03-31T02:00:00Z
```
For comparison:
```
jq-1.5 -nr '1553997600 as $d|$d|todateiso8601'
2019-03-31T02:00:00Z
```

**Comment by jesperoman** [comment]
Ok, found that this was fixed in commit ```3c5b141```, well, just need to compile a more recent binary and distribute locally... :wink:

**Comment by 0wnrepo** [comment]
> Ok, found that this was fixed in commit `3c5b141`, well, just need to compile a more recent binary and distribute locally... 😉

Any ideas how to compare dates without fromdateiso8601 ?

**Comment by jesperoman** [comment]
> > Ok, found that this was fixed in commit `3c5b141`, well, just need to compile a more recent binary and distribute locally... 😉
> 
> Any ideas how to compare dates without fromdateiso8601 ?

Not sure how you mean?
If date is in iso8601 it's probably best to use fromdateiso8601, if not, I guess you need to parse it manually with some regex or something, depending on how your comparison should work?

**Comment by PenelopeFudd** [comment]
We're coming up on the 3-year anniversary of jq 1.6; I could be happy if jq 1.6.0.1 came out with nothing more than this commit included.  :-)

I thought I was clever when I came up with this offset calculation:
```
$ jq -rn '((now | strflocaltime("%s") | tonumber) - now | ceil)'

3600
```
But then realized that I wasn't as clever as I thought:
```
$ jq -rn '0 | strflocaltime("%s")'

0

$ jq -rn '10000000 | strflocaltime("%s") | tonumber - 10000000'

3600
```
The offset depends on Daylight Savings Time, and so my oh-so-fiendishly-clever offset calculation is only good for timestamps in the same part of the year (with/without DST).  Gah!

This overcomes that, but looks atrocious:
```
$ jq -rn '( "2021-09-16T04:16:19Z" | fromdate ) as $a | ($a | strflocaltime("%s") | tonumber - $a ) as $offset | [$a,$a - $offset]'
[
  1631769379,
  1631765779
]
```
"Happy birthday, dear JQ!"  :-)
